### PR TITLE
feat: infrastructure agent network issues troubleshooting

### PR DESCRIPTION
**Pull request by josemore**
## Give us some context

* Adding concrete network troubleshooting steps when dealing with connectivity issues from the host running the infrastructure agent and New Relic endpoints.

### CODE REVIEW

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+## Problem
+
+The infrastructure agent is experiencing network issues connecting to New Relic endpoints.
+Connectivity errors such as "Collector endpoint not reachable" or "Timeout exceeded while awaiting headers" appear on the agent logs.
+
+## Solution
+
+Verify that the New Relic infrastructure agent endpoints are reachable from the host where it's installed.
+Follow the instructions below to confirm there are no issues with your environment's network settings:
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:20]:
```suggestion
The infrastructure agent is experiencing network issues connecting to New Relic endpoints.
Connectivity errors such as `Collector endpoint not reachable` or `Timeout exceeded while awaiting headers` appear on the agent logs.

## Solution

Verify that the New Relic infrastructure agent endpoints are reachable from the host where it's installed.
Follow the instructions below to confirm there are no issues with your environment's network settings:
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+<CollapserGroup>
+  
+<Collapser id="windows" title="Windows">
+
+Run the following command on the terminal:
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:27]:
```suggestion
Run the following command on the terminal:

```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+$ Invoke-WebRequest -Uri "https://infra-api.newrelic.com/cdn-cgi/trace"
+```
+
+The expected output should include a `200 (OK)` as shown below:
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:32]:
```suggestion
The expected output should include a `200 (OK)` as shown below:

```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+RawContentLength  : 284
+```
+
+</Collapser>
+
+<Collapser id="linux" title="Linux">
+
+1. **Resolve the domain:** Use `nslookup` or `dig` to validate that the Linux system can resolve the domain to an IP address:
+
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:67]:
```suggestion
```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+$ nslookup infra-api.newrelic.com
+```
+
+The output should contain a name and IP address.
+
+2. **Ping the server:** Try pinging the IP address obtained above to ensure you can reach the server.
+
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:75]:
```suggestion
```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+$ ping 162.247.241.2
+```
+
+The server should be reachable.
+
+3. **Traceroute:** Use traceroute to check the network path between your system and the server:
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:81]:
```suggestion
3. **Traceroute:** Use `traceroute` to check the network path between your system and the server:
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:83]:
```suggestion
```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+$ traceroute -I infra-api.newrelic.com
+```
+
+Analyze the output to identify potential network issues, such as high latency or packet loss.
+
+4. **Curl:** Use `curl` to check if the endpoint is reachable:
+
+```bash
+$ curl -v https://infra-api.newrelic.com/cdn-cgi/trace
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:92]:
```suggestion
curl -v https://infra-api.newrelic.com/cdn-cgi/trace
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+```
+$ ping 162.247.241.2
+```
+
+The server should be reachable.
+
+3. **Traceroute:** Use traceroute to check the network path between your system and the server:
+
+```
+$ traceroute -I infra-api.newrelic.com
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:84]:
```suggestion
traceroute -I infra-api.newrelic.com
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+```
+$ nslookup infra-api.newrelic.com
+```
+
+The output should contain a name and IP address.
+
+2. **Ping the server:** Try pinging the IP address obtained above to ensure you can reach the server.
+
+```
+$ ping 162.247.241.2
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:76]:
```suggestion
ping 162.247.241.2
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+```
+
+</Collapser>
+
+<Collapser id="linux" title="Linux">
+
+1. **Resolve the domain:** Use `nslookup` or `dig` to validate that the Linux system can resolve the domain to an IP address:
+
+```
+$ nslookup infra-api.newrelic.com
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:68]:
```suggestion
nslookup infra-api.newrelic.com
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+Verify that the New Relic infrastructure agent endpoints are reachable from the host where it's installed.
+Follow the instructions below to confirm there are no issues with your environment's network settings:
+
+<CollapserGroup>
+  
+<Collapser id="windows" title="Windows">
+
+Run the following command on the terminal:
+```
+$ Invoke-WebRequest -Uri "https://infra-api.newrelic.com/cdn-cgi/trace"
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:28]:
```suggestion
Invoke-WebRequest -Uri "https://infra-api.newrelic.com/cdn-cgi/trace"
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+
+## Troubleshoot with NRDiag
+
+New Relic diagnostics tool `NRDiag` is a utility that can also be used to automatically detects network problems.
+Follow these steps to run the infrastructure agent connection test:
+
+1. Follow the [getting started](https://docs.newrelic.com/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/diagnostics-cli-nrdiag/#get-started) steps to install `NRDiag` on your host.
+2. Run the infrastructure agent connect test.
+
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:125]:
```suggestion
```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+./nrdiag -t Infra/Agent/Connect
+```
+
+3. Validate the results, the expected output should look like the report below:
+
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:131]:
```suggestion
```bash
```

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx (hunk 12 above)
Comment (ally-sassman) [reply src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:28]: I removed all the $ signs so it doesn't interrupt the user flow of copying/pasting the code from the doc.

File: src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx (hunk 3 above)
Comment (ally-sassman) [reply src/content/docs/infrastructure/new-relic-infrastructure/troubleshooting/network-connection-issues.mdx:32]: @josemore Should the output be in JSON or bash format?

**Review by akristen — APPROVED** [review]
Looks good!

**Pull request merged**